ocaml/sdk-gen/go/test_gen_go.ml

```diff
@@ -0,0 +1,311 @@
+(* Copyright (C) Cloud Software Group, Inc.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License as published
+   by the Free Software Foundation; version 2.1 only. with the special
+   exception on linking described in file LICENSE.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU Lesser General Public License for more details.
+*)
+
+open Test_highlevel
+open CommonFunctions
+open Gen_go_helper
+
+let test_data_dir = "test_data"
+
+let string_of_file filename =
+  string_of_file (test_data_dir // filename) |> String.trim
+
+module SnakeToCamelTest = Generic.MakeStateless (struct
+  module Io = struct
+    type input_t = string
+
+    type output_t = string
+
+    let string_of_input_t = Test_printers.string
+
+    let string_of_output_t = Test_printers.string
+  end
+
+  let transform = snake_to_camel
+
+  let tests =
+    `QuickAndAutoDocumented
+      [
+        ("ni_hao-Nanjin", "NiHaoNanjin")
+      ; ("ni_hao", "NiHao")
+      ; ("nanjing", "Nanjing")
+      ]
+end)
+
+let rec is_same_struct_of_value (value1 : Mustache.Json.value)
```

Comment:
This seems a JSON schema checking.
I would suggest either defining records for the JSON schemas, or the following:
```
let schema_check keys checker members =
  let keys' = List.map (fun k, _ -> k) members in
  (compare_keys keys keys') && (List.for_all verify_module_member members)

(* field *)
let verify_field_member = function
  | "name", `String _
  | "description", `String _
  | "type", `String _  ->
      true
  | _ ->
      false

let field_keys = ["name"; "description"; "type"]

let verify_field = function
  | `O members ->
      schema_check field_keys verify_field_member members
  | _ ->
      false

(* module *)
let verify_module_member = function
  | "name", `String _ ->
      true
  | "sname", `Null ->
      true
  | _ ->
      false

let module_keys = ["name"; "sname"]

let verify_modules_item = function
  | `O members ->
      schema_check module_keys verify_module_member members
  | _ ->
      false

(* modules *)
let modules_keys = ["import"; "items"]

let verify_modules_member = function
  | "import", `Bool _ ->
      true
  | "items", `A items ->
      List.for_all verify_modules_item items
  | _ ->
      false

(* enums *)
let verify_enum_member = function
  | "value", `String _
  | "doc", `String _
  | "name", `String _
  | "type", `String _ ->
      true
  | _ ->
      false

let enum_keys = ["value"; "doc"; "name"; "type"]

let verify_enum = function
  | `O members ->
      schema_check enum_keys verify_enum_member members
  | _ ->
      false

(* obj *)
let verify_obj_member = function
  | "name", `String _
  | "description", `String _ ->
      true
  | "fields", `A fields ->
      List.for_all verify_field fields
  | "enums", `A enums ->
      List.for_all verify_enum enums
  | "modules", `O members ->
      schema_check modules_keys verify_modules_member members
  | _ ->
      false
in

let obj_keys = ["name"; "description"; "fields"; "enums"; "modules"]

let verify_obj = function
  | `O members ->
      schema_check obj_keys verify_obj_member members
  | _ ->
      false
```